app/controllers/list_controller.rb

```diff
@@ -257,7 +257,11 @@ def top(options = nil)
     options ||= {}
     period = params[:period]
     period ||= ListController.best_period_for(current_user.try(:previous_visit_at), options[:category])
-    public_send("top_#{period}", options)
+    begin
+      public_send("top_#{period}", options)
+    rescue NoMethodError
```

Comment:
The trouble with this is that a `NoMethodError` may be raised during the course of executing the method. How about checking with `respond_to?` first?